### PR TITLE
Gtfs reader - reading agency for networks

### PR DESCRIPTION
**Pull request by prhod**

### CODE REVIEW

File: examples/gtfs_reader.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+extern crate csv;
+extern crate navitia_model;
+
+fn main() {
+    let objects = navitia_model::gtfs::read("./examples/data/");
+    println!("Count of networks loaded : {}", objects.networks.len());
```
Comment (TeXitoi) [inline examples/gtfs_reader.rs]: you can dump the model using serde_json

File: examples/gtfs_reader.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+extern crate csv;
+extern crate navitia_model;
+
+fn main() {
+    let objects = navitia_model::gtfs::read("./examples/data/");
```
Comment (TeXitoi) [inline examples/gtfs_reader.rs]: better to read current directory

File: src/gtfs.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    "1".to_string()
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct Agency {
+    #[serde(rename = "agency_id", default = "default_agency_id")] id: String,
+    #[serde(rename = "agency_name")] name: String,
+    #[serde(rename = "agency_url")] url: String,
+    #[serde(rename = "agency_timezone")] timezone: Option<String>,
+    #[serde(rename = "agency_lang", default)] lang: Option<String>,
```
Comment (TeXitoi) [inline src/gtfs.rs]: I don't think you need default for option fields.

File: src/gtfs.rs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    #[serde(rename = "agency_phone", default)] phone: Option<String>,
+    #[serde(rename = "agency_fare_url", default)] fare_url: Option<String>,
```
Comment (TeXitoi) [inline src/gtfs.rs]: If there is field you don't use, just don't put them.

File: tests/gtfs_reader.rs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+use std::io::prelude::*;
+
+#[test]
+fn load_minimal_agency() {
+    let agency_content = "agency_name,agency_url,agency_timezone\n
+    My agency,http://my-agency_url.com,Europe/London";
+    let tmp_dir = TempDir::new("osm_transit_extractor").expect("create temp dir");
+    let file_path = tmp_dir.path().join("agency.txt");
+    let mut f = File::create(&file_path).unwrap();
+    f.write_all(agency_content.as_bytes()).unwrap();
```
Comment (TeXitoi) [inline tests/gtfs_reader.rs]: better to add the files in the repo and read them.

File: tests/gtfs_reader.rs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+use navitia_model::gtfs;
+use tempdir::TempDir;
+use std::fs::File;
+use std::io::prelude::*;
+
+#[test]
+fn load_minimal_agency() {
+    let agency_content = "agency_name,agency_url,agency_timezone\n
+    My agency,http://my-agency_url.com,Europe/London";
+    let tmp_dir = TempDir::new("osm_transit_extractor").expect("create temp dir");
```
Comment (TeXitoi) [inline tests/gtfs_reader.rs]: That's not osm_transit_extractor ;-)

Comment (prhod) [reply tests/gtfs_reader.rs]: dammit, I've been spotted :p

File: tests/gtfs_reader.rs (hunk 5 above)
Comment (prhod) [reply tests/gtfs_reader.rs]: I started with a fixture, but it led to an awful list of subdirectories. I thought it best to create the file one the fly for those simple tests.

File: examples/gtfs_reader.rs (hunk 1 above)
Comment (prhod) [reply examples/gtfs_reader.rs]: I don't understand what you want me to do here. Print out a JSON output for the first network ?

Comment (TeXitoi) [reply examples/gtfs_reader.rs]: That's for you. I don't really care. You can easilly dump the model as json, interesting for testing.

Comment (prhod) [reply examples/gtfs_reader.rs]: ok, good idea. I'll try to use it in the tests. If you're ok with it, I'll let the example like that for the moment.

Comment (TeXitoi) [reply examples/gtfs_reader.rs]: In test, that's easier to directly check the structure.

As you want, that's just for you ;-)

File: tests/gtfs_reader.rs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    f.write_all(agency_content.as_bytes()).unwrap();
+
+    let networks = gtfs::read_agency(tmp_dir.path());
+    tmp_dir.close().expect("delete temp dir");
+    assert_eq!(1, networks.len());
+}
+
+#[test]
+fn load_complete_agency() {
+    let agency_content = "agency_id,agency_name,agency_url,agency_timezone,agency_lang,agency_phone,agency_fare_url,agency_email\n
```
Comment (pbougue) [inline tests/gtfs_reader.rs]: Not sure (and for knowledge only), but you might use https://doc.rust-lang.org/reference/tokens.html#raw-string-literals

**Pull request merged**